src/utils/NewsApi.js

```diff
@@ -0,0 +1,43 @@
+import { fromDate, toDate } from "./helpers";
+
+const BASE_URL = "https://nomoreparties.co/news/v2/everything?";
+const API_KEY = "397db29f2b7d47238daa4bdcefcbdfd2";
+
+export class NewsApi {
+  constructor(baseUrl, apiKey, fromDate, toDate) {
+    this.apiKey = apiKey;
+    this.baseUrl = baseUrl;
+    this.date = { from: fromDate, to: toDate };
+  }
+
+  getArticleData = (query) => {
+    return fetch(
+      `${this.baseUrl}q=${query}&language=en&from=${this.date.from}&to=${this.date.to}&pageSize=100&apiKey=${this.apiKey}`
+    )
+      .then((res) => {
```

Comment:
* you need to use `checkResponse` for checking responses not to duplicate the code